src/easier68k/core/opcodes/dc.py

```diff
@@ -0,0 +1,392 @@
+                                hexed = '0' + hexed
+
+                        if size == 'W':
+                            while len(hexed) % 4 != 0:
+                                hexed = '0' + hexed
+
+                        if size == 'B':
+                            while len(hexed) % 2 != 0:
+                                hexed = '0' + hexed
+
+                        # Length is now for sure even
+                        for i in range(0, len(hexed), 2):
+                            params.append('$' + hexed[i:i+2])
+
+                is_string = False
+                quote_delim = None
+                current_param = ''
+                continue
+
+            if quote_delim and c == quote_delim:  # Breaking out of the quote
+                quote_delim = None
+                continue
+
+            if c == '\'':
+                if not quote_delim:  # This is the start of a quote
+                    quote_delim = '\''
+                    current_param = ''
+                    is_string = True
+                    continue
+
+            if c == '\"':
+                if not quote_delim:  # This is the start of a quote
+                    quote_delim = '\"'
+                    current_param = ''
+                    is_string = True
+                    continue
+
+            current_param += c
+
+        if current_param:
+            if is_string:
+                temp_params = []
+                for c in current_param:
+                    temp_params.append('$' + format(ord(c), "x"))
+
+                # Right pad the string with zeroes
+                if size == 'L':
+                    while len(temp_params) % 4 != 0:
+                        temp_params.append('$00')
+                if size == 'W':
+                    while len(temp_params) % 2 != 0:
+                        temp_params.append('$00')
+
+                params.extend(temp_params)
+            else:
+                # Make the value the right hex length
+                val = parse_literal(current_param.strip())
+                hexed = hex(val)[2:].upper()
+                if size == 'L':
+                    while len(hexed) % 8 != 0:
+                        hexed = '0' + hexed
+
+                if size == 'W':
+                    while len(hexed) % 4 != 0:
+                        hexed = '0' + hexed
+
+                if size == 'B':
+                    while len(hexed) % 2 != 0:
+                        hexed = '0' + hexed
+
+                # Length is now for sure even
+                for i in range(0, len(hexed), 2):
+                    params.append('$' + hexed[i:i+2])
+
+        return DC(params, size), issues
+
+    @staticmethod
+    def is_valid(command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        >>> DC.is_valid('DC.B', '\\'Hello, world!\\'')[0]
+        True
+
+        >>> DC.is_valid('DC.B', '\\'Hello\\'\\' world!\\'')[0]
```

Comment:
This is where I would give up on doctests and start using pytest. This is good, but maybe in future it would be more readable to do this in a pytest. Escaping looks to be kinda tricky.

Reply:
Escaping is super weird, yeah: I don't think that'll come up anywhere else though since DC is (afaik) the only place that uses string literals (?).